cli/src/commands/split.rs

```diff
@@ -179,16 +174,33 @@ the operation will be aborted.
     // commit.
     tx.mut_repo()
         .set_rewritten_commit(commit.id().clone(), second_commit.id().clone());
-    let num_rebased = if args.siblings {
-        rebase_children_for_siblings_split(
-            &mut tx,
-            command.settings(),
-            &commit,
-            vec![first_commit.clone(), second_commit.clone()],
-        )?
-    } else {
-        tx.mut_repo().rebase_descendants(command.settings())?
-    };
+    let mut num_rebased = 0;
+    tx.mut_repo().transform_descendants(
+        command.settings(),
+        vec![commit.id().clone()],
+        |mut rewriter| {
+            let old_parents = rewriter.old_commit().parent_ids();
+            // Reserve space up front since we know we're adding at most one more parent.
+            let mut new_parents = Vec::with_capacity(old_parents.len() + 1);
+            for parent in old_parents {
+                if parent != commit.id() {
+                    new_parents.push(parent.clone());
+                    continue;
+                }
+                if args.siblings {
+                    new_parents.push(first_commit.id().clone());
+                }
+                new_parents.push(second_commit.id().clone());
+                num_rebased += 1;
```

Comment:
Shouldn't this be updated outside the `for parent in old_parents` loop?

Reply:
Actually yeah, I think so. I misunderstood how the count worked previously. I pushed another commit with testing changes that would have caught this.